src/MicrosoftTeams.ts

```diff
@@ -417,26 +428,52 @@ namespace microsoftTeams
         {
             ensureInitialized(frameContexts.content, frameContexts.settings, frameContexts.remove);
 
+            authParams = authenticateParameters;
+
             // Convert any relative URLs into absolute ones before sending them over to our parent window
             let link = document.createElement("a");
-            link.href = authenticateParameters.url;
-
-            let messageId = sendMessage(
-                "authentication.authenticate",
-                link.href,
-                authenticateParameters.width,
-                authenticateParameters.height);
-            callbacks[messageId] = (success: boolean, response: string) =>
+            link.href = authParams.url;
+
+            // Open a popup window to the specified authentication URL
+            if (loadedInDesktopClient)
+            {
+                let messageId = sendMessage("authentication.authenticate", link.href);
+                callbacks[messageId] = (allowed: boolean) =>
+                {
+                    if (allowed)
+                    {
+                        authWindow = currentWindow.open(link.href, "Microsoft Teams", "width = " + authParams.width + ", height = " + authParams.height);
+                    }
+                    else
+                    {
+                        authParams = null;
+                        throw new Error("Authentication is only supported for URLs matching the pattern registered in the manifest.");
+                    }
+                };
+            }
+            else
+            {
+                authWindow = currentWindow.open(link.href, "Microsoft Teams", "width = " + authParams.width + ", height = " + authParams.height);
+            }
+
+            if (!authWindow)
+            {
+                authParams = null;
+                throw new Error("Failed to open authentication window.");
+            }
+
+            // Wait for the authentication window to close
+            authInterval = setInterval(() =>
             {
-                if (success)
+                if (!authWindow || authWindow.closed)
                 {
-                    authenticateParameters.successCallback(response);
+                    handleFailure();
                 }
```

Comment:
Is there a reason we can include to the caller to help them understand what happened?